pkgs/by-name/pr/protege/package.nix

```
@@ -0,0 +1,104 @@
+{ stdenvNoCC
+, lib
+, fetchFromGitHub
+, copyDesktopItems
+, iconConvTools
+, makeDesktopItem
+, makeWrapper
+, jdk11
+, maven
+}:
+
+let
+  pname = "protege";
+  version = "5.6.4";
+  mvn = maven.override { jdk = jdk11; };
+
+  # Build a platform-independent core from which we can derive platform-specific
+  # directory layouts, so that optimized binary caches can de-duplicate it
+  # effectively.
+  protege-build = mvn.buildMavenPackage {
+    pname = "${pname}-build";
+    inherit version;
+
+    src = fetchFromGitHub {
+      owner = "protegeproject";
+      repo = pname;
+      rev = version;
+      hash = "sha256-Q3MHa7nCeF31n7JPltcemFBc/sJwGA9Ev0ymjQhY/U0=";
+    };
+
+    patches = [
+      ./enforce-plugin-versions.patch
+      ./platform-independent-only.patch
+    ];
+
+    mvnHash = "sha256-GY9qWqRjNQJMd/hoX0S5HIoYtE5pOceZoJRPTkogcao=";
+
+    installPhase = ''
+      runHook preInstall
+
+      mv protege-desktop/target/protege-${version}-platform-independent/Protege-${version} $out
+
+      runHook postInstall
+    '';
+  };
+in
+stdenvNoCC.mkDerivation rec {
+  inherit pname version;
+
+  buildInputs = [ protege-build ];
+  nativeBuildInputs = [
+    copyDesktopItems
+    iconConvTools
+    jdk11
+    makeWrapper
+  ];
+
+  dontUnpack = true;
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/share/${pname}
+
+    # Place a wrapper for the launcher script into a default /bin location
+    makeWrapper ${protege-build}/run.sh $out/bin/${pname} \
+      --set JAVA_HOME ${jdk11.home}
+
+    # Link all jars from within the standard /share/${pname} directory
+    ln -s -t $out/share/${pname} ${protege-build}/bundles/*
+
+    # Generate and copy icons to where they can be found
+    icoFileToHiColorTheme ${protege-build}/app/Protege.ico ${pname} $out
+
+    runHook postInstall
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      desktopName = "Protege Desktop";
+      genericName = "Ontology Editor";
+      icon = pname;
+      comment = meta.description;
+      categories = [ "Development" ];
+      exec = pname;
+    })
+  ];
+
+  meta = with lib; {
+    homepage = "https://protege.stanford.edu/";
+    downloadPage = "https://protege.stanford.edu/software.php#desktop-protege";
+    description = "A free and open-source OWL 2 ontology editor";
+    longDescription = ''
+      Protégé Desktop is a feature rich ontology editing environment with full support for the OWL 2 Web Ontology Language, and direct in-memory connections to description logic reasoners.
+    '';
+    maintainers = with maintainers; [ nessdoor ];
+    license = with licenses; [ bsd2 ];
+    platforms = platforms.linux;
+    mainProgram = pname;
+    sourceProvenance = with sourceTypes; [ fromSource binaryBytecode ];
```

```suggestion
    maintainers = with maintainers; [ nessdoor ];
    license = with lib.licenses; [ bsd2 ];
    mainProgram = "protege";
    sourceProvenance = with lib.sourceTypes; [ fromSource binaryBytecode ];
```

AFAICR `platforms` is set automatically.

---

I wanted to be explicit about the fact that having this package work only on Linux is a deliberate choice, as I wrote on the PR. Should I add a comment about it over the `platforms` line, as a reminder that this is not a technical limitation?